File: files/2026-05-31/CSEC 31052026.R
library(tidyverse)
library(readxl)

path <- "2026-05-31/Challenge 128.xlsx"
input <- read_excel(path, range = "B2:B8")
test <- read_excel(path, range = "D2:E8")

result <- input %>%
  mutate(
    word = str_extract_all(Sentence, "\\b\\w+\\b"),
    `Longest Word` = map_chr(word, ~ .x[which.max(nchar(.x))]),
    `Length` = nchar(`Longest Word`)
  ) %>%
  select(`Longest Word`, Length)

all.equal(result, test)
# [1] TRUE

File: files/2026-05-31/CSEC 31052026.py
import pandas as pd
import re

path = "2026-05-31/Challenge 128.xlsx"
input = pd.read_excel(path, usecols="B", nrows=7, skiprows=1)
test = pd.read_excel(path, usecols="D:E", nrows=7, skiprows=1)

result = (
	input.copy()
	.pipe(lambda df: df.assign(word=df['Sentence'].apply(lambda x: re.findall(r'\b\w+\b', str(x)))))
	.assign(
		**{
			'Longest Word': lambda df: df['word'].apply(lambda x: max(x, key=len) if x else ''),
			'Length': lambda df: df['Longest Word'].apply(len),
		}
	)
	[['Longest Word', 'Length']]
)

print(result.equals(test))
# True